extensions/wikia/UserRenameTool/RenameUserProcess.class.php

```diff
@@ -174,32 +175,67 @@ public function setup() {
 			$oNewSpoofUser = new SpoofUser( $nun );
 			$conflicts = $oNewSpoofUser->getConflicts();
 			if ( !empty( $conflicts ) ) {
-				$this->addError( wfMessage( 'userrenametool-error-antispoof-conflict', $nun ) );
-				return false;
+				$errors[] = wfMessage( 'userrenametool-error-antispoof-conflict', $nun );
 			}
 
 			// SUS-4301 | check for emojis in user name
 			if ( $oNewSpoofUser->isLegal() === false ) {
-				$this->addError( wfMessage( 'userrenametool-error-antispoof-conflict', $nun ) );
-				return false;
+				$errors[] = wfMessage( 'userrenametool-error-antispoof-conflict', $nun );
 			}
 		} else {
-			$this->addError( wfMessage( 'userrenametool-error-antispoof-notinstalled' ) );
-			return false;
+			$errors[] = wfMessage( 'userrenametool-error-antispoof-notinstalled' );
 		}
 
 		// Phalanx test
 
-		$error = self::testBlock( $oun );
-		if ( !empty( $error ) ) {
-			$this->addError( $error );
-			return false;
+		$err = self::testBlock( $oun );
+		if ( !empty( $err ) ) {
+			$errors[] = $err;
 		}
 
-		$error = self::testBlock( $nun );
-		if ( !empty( $error ) ) {
-			$this->addError( $error );
-			return false;
+		$err = self::testBlock( $nun );
+		if ( !empty( $err ) ) {
+			$errors[] = $err;
+		}
+		if( count( $errors ) > 0 && $stringErrors ){
+			$stringErr = [];
+			foreach( $errors as $err ){
+				$stringErr[] = (string)$err;
```

Comment:
Casting to string here shouldn't be needed as long as the `wfMessage` calls explicitly state their output format (otherwise `parse` is assumed). And as a best practice, we should always specify an explicit output format with `wfMessage`.